### PR TITLE
deprecate in-house k8s client

### LINKED ISSUE #296: Deprecate in-house K8s client with the official K8s client
- service discovery
- configuration discovery

### PR DESCRIPTION
**Pull request by pgressa**
Fixes #296

### CODE REVIEW

**Comment by sdelamo** [comment]
We have migrated the internals to use Project Reactor instead of RxJava2.  This PR should merge master into it.

File: kubernetes-discovery-client/src/main/java/io/micronaut/kubernetes/configuration/KubernetesConfigurationClient.java [hunk 1]
```diff
@@ -66,19 +72,21 @@
...
     private final KubernetesConfiguration configuration;
 
     /**
-     * @param client        An HTTP Client to query the Kubernetes API.
+     * @param client        An Core HTTP Client to query the Kubernetes API.
      * @param configuration The configuration properties
      */
-    public KubernetesConfigurationClient(KubernetesClient client, KubernetesConfiguration configuration) {
+    public KubernetesConfigurationClient(CoreV1ApiReactorClient client, KubernetesConfiguration configuration) {
+        LOG.info("MRDKAAAAAAAAAAAAAA");
```
Comment (rdesgroppes) [inline kubernetes-discovery-client/src/main/java/io/micronaut/kubernetes/configuration/KubernetesConfigurationClient.java:89]: @pgressa oops?

Comment (sdelamo) [reply kubernetes-discovery-client/src/main/java/io/micronaut/kubernetes/configuration/KubernetesConfigurationClient.java:89]: @rdesgroppes thanks for reporting. I have removed the snippet: 

https://github.com/micronaut-projects/micronaut-kubernetes/commit/2bbd37906eef70f838d3a653ba098492e65b870b

Comment (pgressa) [reply kubernetes-discovery-client/src/main/java/io/micronaut/kubernetes/configuration/KubernetesConfigurationClient.java:89]: @rdesgroppes thanks!

**Pull request merged**